src/CommunityToolkit/Xamarin.CommunityToolkit/Behaviors/Validators/ValidationBehavior.shared.cs

```diff
@@ -291,12 +295,28 @@ async ValueTask UpdateStateAsync(bool isForced, CancellationToken? parentToken =
 
 		void UpdateStyle()
 		{
-			if (View == null || (ValidStyle ?? InvalidStyle) == null)
+			if (View == null)
+			{
 				return;
+			}
+
+			if ((ValidStyle ?? InvalidStyle) != null)
+			{
+				View.Style = IsValid
+					? ValidStyle
+					: InvalidStyle;
+				return;     // either use VSM or styles?
+			}
 
-			View.Style = IsValid
-				? ValidStyle
-				: InvalidStyle;
+			var vsmGroupsList = VisualStateManager.GetVisualStateGroups(View);
+			if (vsmGroupsList.Count > 0)
+			{
+				var vsmGroupsstates = vsmGroupsList[0].States;
+				if (vsmGroupsstates.All(x => x.Name.Equals(ValidVisualState) || x.Name.Equals(UnValidVisualState)))
+				{
```

Comment:
Do we need these checks? is it ok to use Linq? Do wee need to set the State to Null/default in some cases?